### PR TITLE
assert messages for std.string

### PR DESCRIPTION
**Pull request by burner**
starting to work on dlang/phobos/project/2

### CODE REVIEW

File: std/string.d [hunk 1]
```diff
@@ -6157,7 +6166,7 @@ C1[] tr(C1, C2, C3, C4 = immutable char)
         if (mod_s && modified && newc == lastc)
             continue;
         result.put(newc);
-        assert(newc != dchar.init);
+        assert(newc != dchar.init, "character not be dchar.init");
```
Comment (thewilsonator) [inline std/string.d]: _must_

**Comment by wilzbach** [comment]
CircleCi failure was due to the PR targetting an old master of Phobos. Rebasing against latest upstream/master helps (just did that for you).

**Comment by wilzbach** [comment]
BTW we have a `.dscanner.ini` file with a check for this (this gives you an overview of all places where to look and also prevents regressions):

```patch
diff --git a/.dscanner.ini b/.dscanner.ini
index 4d33cde60..89f63d64c 100644
--- a/.dscanner.ini
+++ b/.dscanner.ini
@@ -196,7 +196,6 @@ assert_without_msg="-etc.c.SQL_,\
 -std.signals,\
 -std.socket,\
 -std.stdio,\
--std.string,\
 -std.traits,\
 -std.typecons,\
 -std.uni,\
```

run with e.g.

```sh
make -f posix.mak dscanner
```

**Comment by burner** [comment]
@wilzbach thank you

**Pull request merged**